packages/react-dom/src/client/ReactDOMFiberOption.js

```diff
@@ -35,12 +37,13 @@ function flattenChildren(children) {
 
 export function validateProps(element: Element, props: Object) {
   // TODO (yungsters): Remove support for `selected` in <option>.
-  if (__DEV__) {
+  if (__DEV__ && !didWarnSelectedSetOnOption) {
```

Comment:
We generally try to avoid `__DEV__ && ...` because it's not always obvious what's static and what's dynamic.
Could you please just use nested ifs?